substratevm/src/com.oracle.svm.agent.core/src/com/oracle/svm/agent/core/HandleSet.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2019, 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.agent.core;
+
+import static com.oracle.svm.core.util.VMError.guarantee;
+import static com.oracle.svm.jni.JNIObjectHandles.nullHandle;
+
+import java.util.Arrays;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.graalvm.nativeimage.c.type.CTypeConversion;
+
+import com.oracle.svm.jni.nativeapi.JNIEnvironment;
+import com.oracle.svm.jni.nativeapi.JNIMethodId;
+import com.oracle.svm.jni.nativeapi.JNIObjectHandle;
+
+public abstract class HandleSet {
+    private final ReentrantLock globalRefsLock = new ReentrantLock();
+    private JNIObjectHandle[] globalRefs = new JNIObjectHandle[16];
```

Comment:
What is 16?

Reply:
The initial number of global references that are tracked. This array can grow dynamically, I think the size was chosen in accordance with the JVMTI per-thread guaranteed local handle capacity. but it could be different? CC @peter-hofer

Reply:
It would be better to make it a const and give it a descriptive name.

Reply:
I agree, I will take care of it.

Reply:
`INITIAL_GLOBAL_REFS_SIZE` or just a comment `// grows` should be fine.